src/components/TaskDetail.jsx

```diff
@@ -104,192 +106,276 @@ export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete, onAddLab
     onUpdate(task.id, { dueDate: date ? date.toISOString() : null });
   };
 
+  const handleAddComment = () => {
+    if (!newComment.trim() || !user) return;
+    
+    const comment = {
+      id: crypto.randomUUID(),
+      text: newComment.trim(),
+      createdAt: new Date().toISOString(),
+      createdBy: user.uid,
+      createdByName: user.email?.split('@')[0] || 'Anonymous'
+    };
+
+    const updatedComments = [...(task.comments || []), comment];
+    onUpdate(task.id, { comments: updatedComments });
+    setNewComment('');
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
// In a new file: useComments.js
export const useComments = (task, onUpdate) => {
  const [newComment, setNewComment] = useState('');
  
  const handleAddComment = useCallback(() => {
    if (!newComment.trim() || !user) return;
    
    const comment = {
      id: crypto.randomUUID(),
      text: newComment.trim(),
      createdAt: new Date().toISOString(),
      createdBy: user.uid,
      createdByName: user.email?.split('@')[0] || 'Anonymous'
    };

    const updatedComments = [...(task.comments || []), comment];
    onUpdate(task.id, { comments: updatedComments });
    setNewComment('');
  }, [newComment, user, task, onUpdate]);

  return { newComment, setNewComment, handleAddComment };
};
```
The comment handling logic should be extracted into a separate custom hook to improve code organization and reusability.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;